File: frontend/src/types/player.ts
import type { PlayerSymbol } from "./game";

export type Player = {
  userId: string;
  username: string;
  rating: number;
  wins: number;
  losses: number;
  draws: number;
  imageUrl: string;
  rank: number;
  createdAt: string;
  updatedAt: string;
};

export type PlayerInfo = {
  userId: string;
  side: PlayerSymbol;
  username: string;
  imageUrl: string;
};

export type RankedPlayer = Player & {
  rank: number;
};

File: services/api-service/app/schemas/player.py
import datetime
import uuid

from app.schemas.base import CustomBaseModel


class PlayerBase(CustomBaseModel):
    username: str
    image_url: str


class PlayerCreate(PlayerBase):
    user_id: uuid.UUID


class PlayerUpdate(CustomBaseModel):
    username: str | None = None
    image_url: str | None = None
    rating: int | None = None
    draws: int | None = None
    losses: int | None = None
    wins: int | None = None


class PlayerRead(PlayerBase):
    user_id: uuid.UUID
    rating: int
    draws: int
    losses: int
    wins: int
    created_at: datetime.datetime
    updated_at: datetime.datetime


class PlayerSummary(PlayerBase):
    user_id: uuid.UUID


class PlayerWithRank(PlayerRead):
    rank: int | None = None
